ui/user_settings.ts

```diff
@@ -135,10 +138,10 @@ function setup_exchange_callback(event) {
                             }
                             // Exchange removal from backend successful
                             enable_key_entries('', 'exchange');
-                            $('#'+exchange_name+'_badge').remove();
-                            var index = settings.connected_exchanges.indexOf(exchange_name);
-                            if (index == -1) {
-                                throw "Exchange " + exchange_name + "was not in connected_exchanges when trying to remove";
+                            $('#' + exchange_name + '_badge').remove();
+                            let index = settings.connected_exchanges.indexOf(exchange_name.toString());
```

Comment:
why add `toString()` here? Isn't is already a string? It's the value of the `#setup_exchange` textbox.

Reply:
Right - the type definitions are ignorant as to the type of the underlying HTML element, i.e. `string | number | string[]`. It makes more sense to cast like `$('#setup_exchange').val() as string`, will fix.